core/modules/replica/Common.h

```diff
@@ -246,11 +281,26 @@ class SqlRequestParams {
 
     bool batchMode = false;
 
+    enum IndexSpec {
+        DEFAULT,
+        UNIQUE,
+        FULLTEXT,
+        SPATIAL
+    };
+    IndexSpec indexSpec = IndexSpec::DEFAULT;
+    std::string indexName;
+    std::string indexComment;
+    std::vector<SqlIndexColumn> indexColumns;
+
     SqlRequestParams() = default;
 
     explicit SqlRequestParams(ProtocolRequestSql const& request);
 
     std::string type2str() const;
+
+    static std::string indexSpec2str(IndexSpec indexSpec);
+    static IndexSpec str2indexSpec(std::string const& str);
+    static ProtocolRequestSql::IndexSpec translateIndexSpec(IndexSpec indexSpec);
```

Comment:
members and functions and enums oh my! 
I'd rather see the members at the bottom and private, but 🤷